### ISSUE TITLE
ArgoCD should support `helm template --skip-tests`

### ISSUE THREAD

**Issue by DerekTBrown** [argoproj/argo-cd#19956]
# Summary
**TL;DR;** Add `source.helm.skip_tests`, which would pass-through the `--skip-tests` flag to `helm template`.

# Motivation
- Helm supports a pattern of placing test manifests in `./templates/tests`. These manifests are included in the output of `helm template` unless the `--skip-tests` flag is passed ([docs](https://helm.sh/docs/helm/helm_template/#:~:text=JSON%20schema%20validation-,%2D%2Dskip%2Dtests,-skip%20tests%20from), [commit](https://github.com/helm/helm/commit/3d66daeb55d947c4d30d542dfb7459afc21a3c10)).
- Users _generally_ do not want to include these manifests in production deployments, since they are merely for testing purposes (ex. https://github.com/kubecost/cost-analyzer-helm-chart/issues/3069).
- Note: As discussed in https://github.com/argoproj/argo-cd/discussions/15302, ArgoCD currently skips manifests that include hooks not supported by ArgoCD, including [helm test hooks](https://helm.sh/docs/topics/chart_tests/). While this feature covers _many_ testing usecases, it is not totally congruent with `--skip-tests`, which excludes everything in the [`/tests` directory](https://github.com/helm/helm/commit/3d66daeb55d947c4d30d542dfb7459afc21a3c10#diff-a15c00615f24e3d585c1e7dee030d66e0e1d1666a34ce1ff038a84dbe2f5dba5R231).

# Proposal
ArgoCD should support passing the `--skip-tests` flag to `helm template` as part of the existing `helm` configuration block. E.g.:

```yaml
source:
  helm:
   ...
   skip_tests: true
```

# How do you think this should be implemented?
- Add `SkipTests` to [`ApplicationSourceHelm`](https://github.com/argoproj/argo-cd/blob/master/pkg/apis/application/v1alpha1/types.go#L367)
- Add `SkipTests` to [`TemplateOpts`](https://github.com/argoproj/argo-cd/blob/master/util/helm/cmd.go#L330).
- Pass `SkipTests` from `ApplicationSourceHelm` to `TemplateOpts` in [`helmTemplate`](https://github.com/argoproj/argo-cd/blob/14a1a552dcdd19671def1fc8a4392a0de8435251/reposerver/repository/repository.go#L1105).
- If `TemplateOpts.SkipTests`, then append `--skip-tests` to the options generated in [`template`](https://github.com/argoproj/argo-cd/blob/master/util/helm/cmd.go#L358).
- (Add requisite unit test coverage for this feature).

**Comment by jaehanbyun** [comment]
Can I try this task?

**Comment by agaudreault** [comment]
It sounds to me like the default behavior of `skipTests` should be true when not specified.

**Comment by jaehanbyun** [comment]
I've implemented the changes to add the source.helm.skip_tests option, which passes through the --skip-tests flag to helm template. However, for this feature to fully work, the helm.sh/hook: test hook needs to be added in the github.com/argoproj/gitops-engine/blob/master/pkg/sync/hook/helm/type.go file.

Since I'm new to contributing to ArgoCD, I might be missing something. If there's a way to address this within the ArgoCD repository, I would appreciate any guidance. Otherwise, I'll proceed by opening an issue and pull request in the GitOps Engine repository to add support for the helm.sh/hook: test hook.

Thank you for your time and assistance!

**Comment by jaehanbyun** [comment]
Thank you for your feedback, @agaudreault.

I initially misunderstood how the `skipTests` option interacts with the `helm.sh/hook: test` annotation. I assumed that setting `skipTests` to `false` would ensure that manifests with the `helm.sh/hook: test` annotation are recognized and deployed. However, upon re-reading the following text, I realize that my understanding was incorrect.

**Comment by jaehanbyun** [comment]
Hello @agaudreault,

While working, I tested the --skip-tests option and expected that the helm template --skip-tests command would exclude all manifests under the tests folder. However, it actually behaves by excluding only the manifests with test-related Helm hooks within the tests folder. Therefore, I believe that the testing use cases currently covered by ArgoCD include those with the --skip-tests option. If I am missing something, could you please let me know?

**Comment by aali309** [comment]
I think this can be closed?
